### PR TITLE
fix(curriculum): clean-up Project Euler 321-340

### PR DESCRIPTION
**Pull request by gikf**
- Cleans description.
- Changes generic `eulerXXX` function name.
- Adds images.
- Uses MathJax, a lot.
- Related to #40896.

Needed before merging:
- [x] Merged PR adding needed images to CDN https://github.com/freeCodeCamp/cdn/pull/92.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-321-swapping-counters.md [hunk 1]
```diff
@@ -8,35 +8,41 @@ dashedName: problem-321-swapping-counters
 
 # --description--
 
-A horizontal row comprising of 2n + 1 squares has n red counters placed at one end and n blue counters at the other end, being separated by a single empty square in the centre. For example, when n = 3.
+A horizontal row comprising of $2n + 1$ squares has $n$ red counters placed at one end and $n$ blue counters at the other end, being separated by a single empty square in the centre. For example, when $n = 3$.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-321-swapping-counters.md:11]:
```suggestion
A horizontal row comprising of $2n + 1$ squares has $n$ red counters placed at one end and $n$ blue counters at the other end, being separated by a single empty square in the center. For example, when $n = 3$.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-321-swapping-counters.md [hunk 2]
```diff
@@ -8,35 +8,41 @@ dashedName: problem-321-swapping-counters
...
+
+<img class="img-responsive center-block" alt="three squares with red and blue counters placed on opposite ends of the row, separated by one empty square" src="https://cdn.freecodecamp.org/curriculum/project-euler/swapping-counters-1.gif" style="background-color: white; padding: 10px;">
 
 A counter can move from one square to the next (slide) or can jump over another counter (hop) as long as the square next to that counter is unoccupied.
 
-Let M(n) represent the minimum number of moves/actions to completely reverse the positions of the coloured counters; that is, move all the red counters to the right and all the blue counters to the left. It can be verified M(3) = 15, which also happens to be a triangle number.
+<img class="img-responsive center-block" alt="allowed moves of the counter" src="https://cdn.freecodecamp.org/curriculum/project-euler/swapping-counters-2.gif" style="background-color: white; padding: 10px;">
+
+Let $M(n)$ represent the minimum number of moves/actions to completely reverse the positions of the coloured counters; that is, move all the red counters to the right and all the blue counters to the left.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-321-swapping-counters.md:19]:
```suggestion
Let $M(n)$ represent the minimum number of moves/actions to completely reverse the positions of the colored counters; that is, move all the red counters to the right and all the blue counters to the left.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-325-stone-game-ii.md [hunk 3]
```diff
@@ -8,39 +8,39 @@ dashedName: problem-325-stone-game-ii
 
 # --description--
 
-A game is played with two piles of stones and two players. At her turn, a player removes a number of stones from the larger pile. The number of stones she removes must be a positive multiple of the number of stones in the smaller pile.
+A game is played with two piles of stones and two players. On each player's turn, the player may removes a number of stones from the larger pile. The number of stones removes must be a positive multiple of the number of stones in the smaller pile.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-325-stone-game-ii.md:11]:
```suggestion
A game is played with two piles of stones and two players. On each player's turn, the player may remove a number of stones from the larger pile. The number of stones removes must be a positive multiple of the number of stones in the smaller pile.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-327-rooms-of-doom.md [hunk 4]
```diff
@@ -10,6 +10,8 @@ dashedName: problem-327-rooms-of-doom
 
 A series of three rooms are connected to each other by automatic doors.
 
+<img class="img-responsive center-block" alt="series of three rooms, connected to each other by automatic doors" src="https://cdn.freecodecamp.org/curriculum/project-euler/rooms-of-doom.gif" style="background-color: white; padding: 10px;">
+
 Each door is operated by a security card. Once you enter a room the door automatically closes and that security card cannot be used again. A machine at the start will dispense an unlimited number of cards, but each room (including the starting room) contains scanners and if they detect that you are holding more than three security cards or if they detect an unattended security card on the floor, then all the doors will become permanently locked. However, each room contains a box where you may safely store any number of security cards for use at a later stage.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-327-rooms-of-doom.md:15]:
```suggestion
Each door is operated by a security card. Once you enter a room, the door automatically closes, and that security card cannot be used again. A machine will dispense an unlimited number of cards at the start, but each room (including the starting room) contains scanners. If they detect that you are holding more than three security cards or if they detect an unattended security card on the floor, then all the doors will become permanently locked. However, each room contains a box where you may safely store any number of security cards for use at a later stage.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-336-maximix-arrangements.md [hunk 5]
```diff
@@ -8,39 +8,47 @@ dashedName: problem-336-maximix-arrangements
 
 # --description--
 
-A train is used to transport four carriages in the order: ABCD. However, sometimes when the train arrives to collect the carriages they are not in the correct order.
+A train is used to transport four carriages in the order: $ABCD$. However, sometimes when the train arrives to collect the carriages they are not in the correct order.
 
-To rearrange the carriages they are all shunted on to a large rotating turntable. After the carriages are uncoupled at a specific point the train moves off the turntable pulling the carriages still attached with it. The remaining carriages are rotated 180 degrees. All of the carriages are then rejoined and this process is repeated as often as necessary in order to obtain the least number of uses of the turntable.
+To rearrange the carriages they are all shunted on to a large rotating turntable. After the carriages are uncoupled at a specific point the train moves off the turntable pulling the carriages still attached with it. The remaining carriages are rotated 180°. All of the carriages are then rejoined and this process is repeated as often as necessary in order to obtain the least number of uses of the turntable.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-336-maximix-arrangements.md:13]:
```suggestion
To rearrange the carriages, they are all shunted onto a large rotating turntable. After the carriages are uncoupled at a specific point, the train moves off the turntable pulling the carriages still attached with it. The remaining carriages are rotated 180°. All of the carriages are then rejoined and this process is repeated as often as necessary in order to obtain the least number of uses of the turntable.
```

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-339-peredur-fab-efrawg.md [hunk 6]
```diff
@@ -8,31 +8,33 @@ dashedName: problem-339-peredur-fab-efrawg
...
 
-Initially each flock consists of n sheep. Each sheep (regardless of colour) is equally likely to be the next sheep to bleat. After a sheep has bleated and a sheep from the other flock has crossed over, Peredur may remove a number of white sheep in order to maximize the expected final number of black sheep. Let E(n) be the expected final number of black sheep if Peredur uses an optimal strategy.
+Initially each flock consists of $n$ sheep. Each sheep (regardless of colour) is equally likely to be the next sheep to bleat. After a sheep has bleated and a sheep from the other flock has crossed over, Peredur may remove a number of white sheep in order to maximize the expected final number of black sheep. Let $E(n)$ be the expected final number of black sheep if Peredur uses an optimal strategy.
```
Comment (Sembauke) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-339-peredur-fab-efrawg.md:13]:
```suggestion
Initially, each flock consists of $n$ sheep. Each sheep (regardless of color) is equally likely to be the next sheep to bleat. After a sheep has bleated and a sheep from the other flock has crossed over, Peredur may remove a number of white sheep in order to maximize the expected final number of black sheep. Let $E(n)$ be the expected final number of black sheep if Peredur uses an optimal strategy.
```

**Review by Sembauke — DISMISSED** [review]

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-328-lowest-cost-search.md [hunk 7]
```diff
@@ -8,45 +8,45 @@ dashedName: problem-328-lowest-cost-search
...
 
-We can improve considerably the worst-case cost for n=8, by asking "5" as our first question. If we are told that the hidden number is higher than 5, our second question will be "7", then we'll know for certain what the hidden number is (for a total cost of 5+7=12). If we are told that the hidden number is lower than 5, our second question will be "3" and if the hidden number is lower than 3 our third question will be "1", giving a total cost of 5+3+1=9. Since 12>9, the worst-case cost for this strategy is 12. That's better than what we achieved previously with the "binary search" strategy; it is also better than or equal to any other strategy. So, in fact, we have just described an optimal strategy for n=8.
+Similarly, $C(100) = 400$ and $\displaystyle\sum_{n = 1}^{100} C(n) = 17575$.
 
-Let C(n) be the worst-case cost achieved by an optimal strategy for n, as described above. Thus C(1) = 0, C(2) = 1, C(3) = 2 and C(8) = 12. Similarly, C(100) = 400 and C(n) = 17575.
-
-Find C(n).
+Find $\displaystyle\sum_{n = 1}^{2\\,000\\,000} C(n)$.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-328-lowest-cost-search.md:29]:
```suggestion
Find $\displaystyle\sum_{n = 1}^{200\\,000} C(n)$.
```

https://projecteuler.net/problem=328

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-331-cross-flips.md [hunk 8]
```diff
@@ -10,37 +10,39 @@ dashedName: problem-331-cross-flips
...
+The bottom left disk on the $N×N$ board has coordinates (0, 0); the bottom right disk has coordinates ($N - 1$,$0$) and the top left disk has coordinates ($0$,$N - 1$).
 
-Let CN be the following configuration of a board with N×N disks: A disk at (x,y) satisfying , shows its black side; otherwise, it shows its white side. C5 is shown above.
+Let $C_N$ be the following configuration of a board with $N × N$disks: A disk at ($x$, $y$) satisfying $N - 1 \le \sqrt{x^2 + y^2} \lt N$, shows its black side; otherwise, it shows its white side. $C_5$ is shown above.
 
-Let T(N) be the minimal number of turns to finish a game starting from configuration CN or 0 if configuration CN is unsolvable. We have shown that T(5)=3. You are also given that T(10)=29 and T(1 000)=395253.
+Let $T(N)$ be the minimal number of turns to finish a game starting from configuration $C_N$ or 0 if configuration $C_N$ is unsolvable. We have shown that $T(5) = 3$. You are also given that $T(10) = 29$ and $T(1\\,000) = 395\\,253$.
 
-Find .
+Find $\displaystyle \sum_{i = 3}^{31} T(2^i - 1)$.
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-331-cross-flips.md:25]:
```suggestion
Find $\displaystyle \sum_{i = 3}^{31} T(2^i - i)$.
```

https://projecteuler.net/problem=331

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-334-spilling-the-beans.md [hunk 9]
```diff
@@ -12,33 +12,43 @@ In Plato's heaven, there exist an infinite number of bowls in a straight line. E
...
-<!-- TODO Use MathJax and re-write from projecteuler.net -->
+<img class="img-responsive center-block" alt="animation of game when two adjacent bowls contains 2 and 3 beans respectivelly" src="https://cdn.freecodecamp.org/curriculum/project-euler/spilling-the-beans.gif" style="background-color: white; padding: 10px;">
 
-You are given the following sequences: t0 = 123456. ti = ti-12, if ti-1 is even ti-12 926252, if ti-1 is odd where ⌊x⌋ is the floor function and is the bitwise XOR operator. bi = ( ti mod 211) + 1.
+You are given the following sequences:
 
-The first two terms of the last sequence are b1 = 289 and b2 = 145. If we start with b1 and b2 beans in two adjacent bowls, 3419100 moves would be required to finish the game.
+$$\begin{align}
+  & t0 = 123456, \\\\
+  & ti = \begin{cases}
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-334-spilling-the-beans.md:21]:
```suggestion
  & t_0 = 123456, \\\\
  & t_i = \begin{cases}
```

https://projecteuler.net/problem=334

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-337-totient-stairstep-sequences.md [hunk 10]
```diff
@@ -8,41 +8,41 @@ dashedName: problem-337-totient-stairstep-sequences
 
 # --description--
 
-Let {a1, a2,..., an} be an integer sequence of length n such that:
+Let $\\{a_1, a_2, \ldots, a_n\\}$ be an integer sequence of length $n$ such that:
 
-a1 = 6
+- $a_1 = 6$
+- for all $1 ≤ i &lt; n$ : $φ(a_i) &lt; φ(a_{i + 1}) &lt; a_i &lt; a_{i + 1}$
```
Comment (moT01) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-337-totient-stairstep-sequences.md:14]:
```suggestion
- for all $1 ≤ i &lt; n$ : $φ(a_i) &lt; φ(a_{i + 1}) &lt; a_i &lt; {a_{i + 1}}^1$
```

https://projecteuler.net/problem=337

**Review by moT01 — COMMENTED** [review]
Lookin good 👍 Found a few discrepancies between the Euler site and these.

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-337-totient-stairstep-sequences.md (hunk 10 above)
Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-337-totient-stairstep-sequences.md:14]: I've moved part to which this `1` references (Euler's totient function) to the next line, so it shouldn't be needed anymore here.

Comment (moT01) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-337-totient-stairstep-sequences.md:14]: ahh - I see it now.

**Review by moT01 — APPROVED** [review]
LGTM 🎉

**Review by naomi-lgbt — APPROVED** [review]

**Pull request merged**